docs/DeveloperGuide.md

```diff
@@ -270,58 +270,208 @@ _{Explain here how the data archiving feature will be implemented}_
+| `*`      | fun-seeking professor/FYP student      | end “stickers” (similar to Telegram stickers) or emoji so that the app                                                                  | is more appealing to use [coming soon]                                        |
+| `*`      | FYP student                            | update my availability slots on the app                                                                                                 | my professor knows when I will be available to meet [coming soon]             |
+| `*`      | well-versed SoC professor              | sort my student’s FYP based on what specialisation they are taking                                                                      | I can better organise the FYPs [coming soon]                                  |
+| `*`      | very meticulous SoC professor          | see student's FYP sorted by research area first, followed by alphabetical order                                                         | I can gain quick easy access to the details required [coming soon]            |
 
 *{More to be added}*
 
+--------------------------------------------------------------------------------------------------------------------
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `JerylFypManager` and the **Actor** is the `Professor`, unless specified otherwise)
+
+**1. Use case: UC01 - Adding FYP**
+
+**Preconditions: Professor knows the details of FYP to be added**
+
+**MSS**
+
+1.  Professor requests to add a student FYP into the list.
+2.  JerylFypManager adds the FYP details.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The request given is invalid.
+  * 1a1. JerylFypManager shows an error message.
+  * 1a2. Professor enters a new request.
+  
+  Steps 1a1-1a2 are repeated until the request entered is correct.
+
+  Use case resumes from step 2.
+
+
+**2. Use case: UC02 - Searching Keyword**
+
+**Preconditions: Professor can recall part of the project name**
+
+**MSS**
+
+1.  Professor wants to track the status of certain project(s).
+2.  Professor searches the project(s) using a specified keyword.
+3.  JerylFypManager displays the search results with a brief summary for each project.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The keyword given does not match any project name.
+    * 2a1. JerylFypManager shows an error message.
+    * 2a2. Professor tries again with another keyword.
+
+  Steps 2a1-2a2 are repeated until some projects are matched.
+
+  Use case resumes from step 3.
+
+
+**3. Use case: UC03 - Marking Project Status**
+
+**Guarantees:**
+* Project Status goes from “Yet to Start” (YTS) to “In Progress”  (IP); (or alternatively, from “In Progress” (IP) to “Done” (DONE).
+
+
+**Preconditions: Existence of said FYP in database**
```

Comment:
Should we remove the term database also? Seems technical

Reply:
`FYP list` would be a more layman term